include/podio/UserDataCollection.h

```diff
@@ -0,0 +1,169 @@
+    void  prepareAfterRead() override final {}
+
+    /// initialize references after read
+    bool setReferences(const ICollectionProvider* ) override final {
+      return true ;
+    }
+
+    /// set collection ID
+    void setID(unsigned id) override final {
+      m_collectionID = id;
+    }
+
+    /// get collection ID
+    unsigned getID() const override final {
+      return m_collectionID ;
+    }
+
+    /// Get the collection buffers for this collection
+    podio::CollectionBuffers getBuffers() override final {
+      return { &_vecPtr,
+	&m_refCollections, // only need to store the ObjectIDs of the referenced objects
+	&m_vecmem_info } ;
+    }
+
+    /// check for validity of the container after read
+    bool isValid() const override final {
+      return true ;
+    }
+
+    /// number of elements in the collection
+    size_t size() const override final {
+      return _vec.size() ;
+    }
+
+    /// fully qualified type name of elements - with namespace
+    std::string getValueTypeName() const override final {
+
+      return UserDataTypes::instance().name( typeid(BasicType) ) ;
+    }
+
+    /// clear the collection and all internal states
+    void clear() override final {
+      _vec.clear() ;
+    } ;
+
+
+    /// check if this collection is a subset collection - no subset possible
+    bool isSubsetCollection() const override final {
+      return false ;
+    }
+
+    /// declare this collection to be a subset collectionv - no effect
+    void setSubsetCollection(bool) override final {}
+
+
+    // ----- some wrapers for std::vector and access to the complete std::vector (if really needed)
+
+    typename std::vector<BasicType>::iterator begin() { return _vec.begin() ; }
+    typename std::vector<BasicType>::iterator end()   { return _vec.end() ; }
+    typename std::vector<BasicType>::const_iterator begin() const { return _vec.begin() ; }
+    typename std::vector<BasicType>::const_iterator end()   const { return _vec.end() ; }
+
+    typename std::vector<BasicType>::reference operator[](size_t idx) { return _vec[idx] ; }
+    typename std::vector<BasicType>::const_reference operator[](size_t idx) const { return _vec[idx] ; }
+
+    void resize( size_t count ) { _vec.resize( count ) ; }
+    void push_back( const BasicType& value ) { _vec.push_back( value ) ; }
+
+    /// access to the actual data vector
+    typename std::vector<BasicType>& vec() {
+      return _vec;
+    }
+
+    /// const access to the actual data vector
+    const typename std::vector<BasicType>& vec() const {
+      return _vec;
+    }
+
+
+  };
+
+  /// some typedefs to make ROOT and the ROOTWriter happy
+
+  typedef UserDataCollection<int>    intCollection ;
+  typedef UserDataCollection<long>   longCollection ;
+  typedef UserDataCollection<float>  floatCollection ;
+  typedef UserDataCollection<double> doubleCollection ;
+
+  typedef int    intData ;
+  typedef long   longData ;
+  typedef float  floatData ;
+  typedef double doubleData ;
```

Comment:
Thanks, Thomas. This seems to do the trick (at least on my centos7 test system) However I do not understand why and more importantly how this can even work, as the branch for the user collection is defined (by the ROOTReader) as `vector<doubleData>` and `doubleData` is a non existing type w/o these typedefs !???